Classes/TextUtils.cs

```diff
@@ -84,14 +86,57 @@ public static string StripDateFromName(string name, DateTime stripDate)
                     // Check if date is within the +/- 'similardateday' range of 'date'
                     if (dateFound >= stripDate.AddDays(-similarDateDayRange) && dateFound <= stripDate.AddDays(similarDateDayRange))
                     {
-                       name = matchStripDate.Replace(name, string.Empty).ToString().Trim();
+                        name = matchStripDate.Replace(name, string.Empty).ToString().Trim();
+                        name = Regex.Replace(name, MatchWSDelimEnd, string.Empty);
                     }
                 }
             }
 
             return name;
         }
 
+        /// <summary>
+        /// Build ‘SmartName’ from the ProgrammeName and EpisodeName values
+        /// Use regex to remove a number of different any duplication of the Programme Name.
+        /// Includes removal of date from episodeName
+        /// </summary>
+        /// <param name="programmeName">Programme Name to be checked.</param>
+        /// <param name="episodeName">Episode Name to be used to form SmartName.</param>
+        /// <param name="stripDate">Date value to be striped from the Episode Name.</param>
+        /// <returns>Episode name with date content removed</returns>
+        public static string BuildEpisodeSmartName(string programmeName, string episodeName, DateTime stripDate)
+        {
+            // RegEx groups
+            const string MatchEpNameEndWSDelim = @"\s(?:\.|\-|\/| |,)(?=\s(?:\.|\-|\/| |,)|$)";
+            const string MatchEpNameDelim = @"(?:\s?)(?:\:|.- +|,+)(?:\s?)";
+
+            // clean episode name of any leading/trailing whitespace and trailing delim characters
+            episodeName = episodeName.Trim();
+            episodeName = Regex.Replace(episodeName, MatchEpNameEndWSDelim, string.Empty);
+
+            // check and remove programme name from episode name (and any delimiting and or white space that follows)
+            episodeName = Regex.Replace(episodeName, "^(?:" + Regex.Escape(programmeName) + ")" + MatchEpNameDelim, string.Empty);
+            episodeName = Regex.Replace(episodeName, "^(?:" + Regex.Escape(programmeName) + ")", string.Empty);
```

Comment:
This should be able to be combined with the line above by making the MatchEpNameDelim part an optional match.